### PR TITLE
Strengthen revert tests

### LINKED ISSUE #28: Check Returned msg in failure tests
Rather than assertFalse check the fallback msg data and verify the data is correct

_Originally posted by @mds1 in https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/27#discussion_r1227205123_

### LINKED ISSUE #29: Rename Revert tests to follow foundry naming convention
_Originally posted by @mds1 in https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/27#discussion_r1226896804_

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Change test names
- Check revert error

Closes #29, closes #28

### CODE REVIEW

File: test/PerpetualPositionRouter.t.sol [hunk 1]
```diff
@@ -38,11 +38,7 @@ contract PositionRouterTest is Test, PerpetualContracts {
     );
     (bool okTwo,) = payable(vethPositionRouterAddr).call(
       abi.encodePacked(
-        uint8(5),
-        uint160(sqrtPriceLimitX96),
-        uint32(block.timestamp),
-        uint96(0),
-        uint96(oppositeAmountBound)
+        uint8(5), uint160(sqrtPriceLimitX96), uint32(block.timestamp), uint96(oppositeAmountBound)
```
Comment (mds1) [inline test/PerpetualPositionRouter.t.sol:41]: _(arbitrary comment location to start a thread)_

PR description says `Closes #29 #28` and it seems github will only recognize the first issue from this, to have it link both you need `Closes #29, closes #28`

File: test/PerpetualPositionRouter.t.sol [hunk 2]
```diff
@@ -222,35 +218,42 @@ contract Fallback is PositionRouterTest {
 
   function testFork_FailedCallWhenExtraCalldataArgument() public {
```
Comment (mds1) [inline test/PerpetualPositionRouter.t.sol:219]: If this PR is intended to close #29, this should be
```suggestion
  function testFork_RevertIf_ExtraCalldataArgument() public {
```

Same for all the other tests that have "Failed" in their name.

Also, should we have an equivalent test for when there's too little calldata?

File: test/PerpetualPositionRouter.t.sol [hunk 3]
```diff
@@ -222,35 +218,42 @@ contract Fallback is PositionRouterTest {
...
+
+    vm.expectRevert(PerpetualPositionRouter.InvalidCalldata.selector);
     (bool ok,) = payable(vethPositionRouterAddr).call(
-      abi.encodePacked(uint8(4), uint160(0), uint32(block.timestamp), uint96(0), uint96(100))
+      abi.encodePacked(
+        uint8(4), uint160(0), uint32(block.timestamp), uint96(10), uint96(100), uint96(0)
+      )
     );
-    assertFalse(ok);
+    assertTrue(ok);
```
Comment (mds1) [inline test/PerpetualPositionRouter.t.sol:228]: Contrary to what the forge book says, I don't think you can use `expectRevert` this way. I changed this to `vm.expectRevert(PerpetualPositionRouter.FunctionDoesNotExist.selector);` and the test still passes.

I think you'll need to delete the expectRevert, change `(book ok,)` to `(bool ok, bytes memory data)` and then `assertEq(data, PerpetualPositionRouter.InvalidCalldata.selector)`

Comment (mds1) [reply test/PerpetualPositionRouter.t.sol:228]: here's an example where the expect revert data is `error message`, but you'd replace that with the `PerpetualPositionRouter.InvalidCalldata.selecto`
```solidity
(bool ok, bytes memory data) = address(a).call(abi.encodeCall(A.foo, ()));
assertFalse(ok, "call should fail");

bytes memory expectedErr = abi.encodeWithSignature("Error(string)", ("error message"));
assertEq(data, expectedErr);
```

Comment (mds1) [reply test/PerpetualPositionRouter.t.sol:228]: Ref https://github.com/foundry-rs/foundry/issues/5159#issuecomment-1591995117

Comment (mds1) [reply test/PerpetualPositionRouter.t.sol:228]: nvm, ignore all of this haha. you're doing things correctly and the book is correct

File: test/PerpetualPositionRouter.t.sol (hunk 2 above)
Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol]: Oops sorry about that, I didn't cherry pick the commit that had the renaming.

Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol]: https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/30/commits/8a01e4d9b39ae82b320a2695cc6cae9ed1b7889c

Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol]: And good call I will add one with not enough calldata. I did try to add one with 0 calldata, but ran into some issues.

Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol]: https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/30/commits/5adf1623ddde21b6a4ff7bd2a7739d47d3333fc1

File: test/PerpetualPositionRouter.t.sol (hunk 1 above)
Comment (alexkeating) [reply test/PerpetualPositionRouter.t.sol:41]: Ah okay thanks!

**Review by mds1 — APPROVED** [review]

**Pull request merged**